core/domain/rights_manager.py

```diff
@@ -403,6 +476,12 @@ def get_collection_rights(
 ) -> Optional[rights_domain.ActivityRights]: ...
 
 
+@overload
+def get_collection_rights(
+    collection_id: str, *, strict: bool = True
+) -> Optional[rights_domain.ActivityRights]: ...
```

Comment:
Should here be more overloads?

Reply:
I think I have added all the cases for overload ( please correct me if I'm wrong somewhere or please provide which case I'm missing ).